testsuite/tests/docstrings/docstrings.ml

```diff
@@ -0,0 +1,575 @@
+    end[@@ocaml.doc " The comment for module Foo "]
+    module type my_module_type  =
+      sig
+        val x : int[@@ocaml.doc " The comment for value x. "]
+        module M :
+        sig val y : int[@@ocaml.doc " The comment for value y. "] end
+        [@@ocaml.doc " The comment for module M. "]
+      end[@@ocaml.doc " The comment for module type my_module_type. "]
+  end =
+  struct
+    let f x y = x + y[@@ocaml.doc " The comment for function f "]
+    [@@@ocaml.text
+      " This comment is not attached to any element since there is another\n      special comment just before the next element. "]
+    exception My_exception of (int -> int) * int
+      [@ocaml.doc
+        " Comment for exception My_exception, even with a simple comment\n      between the special comment and the exception."]
+    type weather =
+      | Rain of int [@ocaml.doc " The comment for constructor Rain "]
+      | Sun [@ocaml.doc " The comment for constructor Sun "][@@ocaml.doc
+                                                              " Comment for type weather  "]
+    type my_record =
+      {
+      foo: int [@ocaml.doc " Comment for field foo "];
+      bar: string [@ocaml.doc " Comment for field bar "]}[@@ocaml.doc
+                                                           " The comment for type my_record "]
+    class my_class =
+      object
+        inherit  cl[@@ocaml.doc
+                     " A comment to describe inheritance from cl "]
+        val mutable tutu = "tutu"[@@ocaml.doc
+                                   " The comment for the instance variable tutu "]
+        val toto = 1[@@ocaml.doc " The comment for toto "]
+        val titi = "titi"[@@ocaml.doc
+                           " Ambiguous comment on both titi and toto "]
+        method toto = tutu ^ "!"[@@ocaml.doc
+                                  " Ambiguous comment on both titi and toto "]
+        method m (f : float) = 1[@@ocaml.doc " The comment for method m "]
+      end[@@ocaml.doc " The comment for class my_class "]
+    class type my_class_type =
+      object
+        val  mutable x : int[@@ocaml.doc
+                              " The comment for the instance variable x. "]
+        method  m : int -> int[@@ocaml.doc " The comment for method m. "]
+      end[@@ocaml.doc " The comment for class type my_class_type "]
+    module Foo =
+      struct
+        external x : int[@@ocaml.doc " The comment for x "][@@ocaml.doc
+                                                             " Another comment for x "]
+      end[@@ocaml.doc " The comment for module Foo "]
+    module type my_module_type  = sig val x : int end[@@ocaml.doc
+                                                       " The comment for module type my_module_type. "]
+  end ;;
+Line _, characters 12-14:
+      inherit cl
+              ^^
+Error: Unbound class cl
+|}]
+
+(***********************************************************************)
+(* Empty doc comments (GPR#548) *)
```

Comment:
Nothing beyond personal preference. I tend to like single test files that act as a form of documentation for how a particular feature behaves. Since I wanted to switch the "empty" tests to use a `-dsource` expect test anyway, I figured I might as well put them with the others.

> And with a single file, would it not be better to merge the docstrings folder with the parsetree one, in order to reduce the number of tests folder in the test suite?

Yeah, that's probably a good idea.